### PR TITLE
docs: update AWS Athena and Redshift docs

### PR DESCRIPTION
**Pull request by mdeshmu**
### SUMMARY
update AWS Athena and Redshift docs

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by rusackas** [comment]
Paging @cbuffevant (via the [Database Familiarity Rolodex](https://docs.google.com/spreadsheets/d/1U1qxiLvOX0kBTUGME1AHHi6Ywel6ECF8xk_Qy-V9R8c/edit#gid=0) - sign on up, everyone!)

**Comment by john-bodley** [comment]
cc @sfirke

**Review by cbuffevant — APPROVED** [review]
LGTM the Readshift part. I tried them out in our cluster.

File: docs/docs/databases/athena.mdx [hunk 1]
```diff
@@ -33,10 +33,8 @@ following connection string:
...
-The PyAthena library also allows to assume a specific IAM role, by [importing the datasource from YAML](https://superset.apache.org/docs/miscellaneous/importing-exporting-datasources/#importing-datasources-from-yaml) and passing extra parameters:
+The PyAthena library also allows to assume a specific IAM role which you can define by adding following parameters in Superset's Athena database connection UI under ADVANCED --> Other --> ENGINE PARAMETERS.
 ```
-databases:
-  - database_name: awsathena
-    sqlalchemy_uri: awsathena+rest://athena.{region_name}.amazonaws.com/{schema_name}?s3_staging_dir={s3_staging_dir}&...
-    extra: "{\"engine_params\": {\"connect_args\": {\"role_arn\": \"{{  ROLE_ARN  }}\" }}}"
+{"connect_args":{"role_arn":"<role arn>"}}
 ```
+
```
Comment (rusackas) [inline docs/docs/databases/athena.mdx:40]:
```suggestion
```
Seems linting doesn't like this extra line

**Comment by rusackas** [comment]
redshift.mdx is failing the linter because of spaces at the ends of lines. If you can clean that up by hand or by running `pre-commit run --all-files` this should pass CI and be mergeable :)

**Comment by mdeshmu** [comment]
Thanks for your attempts to fix it @rusackas :)
I have fixed it. All tests are green now.

File: docs/docs/databases/redshift.mdx [hunk 2]
```diff
@@ -18,8 +20,47 @@ You'll need to the following setting values to form the connection string:
...
+and SQLALCHEMY URI should be set to `redshift+redshift_connector://`
+
+
+#### Using IAM-based credentials with Redshift serverless:
+
+[Redshift serverless](https://docs.aws.amazon.com/redshift/latest/mgmt/serverless-whatis.html) supports connection using IAM roles.
+
+Your superset app's IAM role should have `redshift-serverless:GetCredentials` and `redshift-serverless:GetWorkgroup` permissions on Redshift serverless workgroup.
+
+You have to define following arguments in Superset's redshift database connection UI under ADVANCED --> Others --> ENGINE PARAMETERS.
```
Comment (sfirke) [inline docs/docs/databases/redshift.mdx:62]: Missing word: define **the** following

File: docs/docs/databases/athena.mdx [hunk 3]
```diff
@@ -33,10 +33,7 @@ following connection string:
 awsathena+rest://{aws_access_key_id}:{aws_secret_access_key}@athena.{region_name}.amazonaws.com/{schema_name}?s3_staging_dir={s3_staging_dir}&...
```
Comment (sfirke) [inline docs/docs/databases/athena.mdx:33]: It won't let me mark it on line 29 but that should be "you can also use **the** PyAthena library" -- maybe that can be fixed in this PR even though it's not an addition

**Review by sfirke — COMMENTED** [review]
I can't speak to the substance of the changes - they look like a nice addition! - but left a few small suggestions where there was a missing word or sentence that didn't read smoothly.

**Comment by mdeshmu** [comment]
thanks @sfirke for the review. I have addressed your comments.

**Review by rusackas — APPROVED** [review]

**Pull request merged**